### PR TITLE
DEV: API for plugins to add post update params and handlers

### PR DESCRIPTION
**Pull request by markvanlan**
Plugins need a way to add attributes to posts via the composer, and do something with the params on the server. This API lets plugins permit params to be passed to `PostRevisor`, and also pass a block that will be called with the post and value of the attribute in the `revise!` method.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -208,6 +208,10 @@ def update
       edit_reason: params[:post][:edit_reason]
     }
 
+    Post.plugin_permitted_update_params.each do |param, _|
```
Comment (eviltrout) [inline app/controllers/posts_controller.rb:211]:
```suggestion
    Post.plugin_permitted_update_params.keys.each do |param|
```
I think this is slightly clearer - you only want the keys so why bother with the `_`?

File: lib/plugin/instance.rb [hunk 2]
```diff
@@ -337,6 +337,12 @@ def add_permitted_post_create_param(name, type = :string)
     end
   end
 
+  def add_permitted_post_update_param(attribute, &block)
+    reloadable_patch do |plugin|
+      ::Post.plugin_permitted_update_params[attribute] = { plugin: plugin, handler: block }
```
Comment (eviltrout) [inline lib/plugin/instance.rb:342]: This overwrites any existing attribute. Is that what you wanted? Two plugins couldn't use the same attribute?

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (markvanlan) [reply app/controllers/posts_controller.rb:211]: agreed

File: lib/plugin/instance.rb (hunk 2 above)
Comment (markvanlan) [reply lib/plugin/instance.rb:342]: I guess we could push this into an array!

Comment (markvanlan) [reply lib/plugin/instance.rb:342]: I followed the pattern for `plugin_permitted_create_params

```
  def add_permitted_post_create_param(name, type = :string)
    reloadable_patch do |plugin|
      ::Post.plugin_permitted_create_params[name] = { plugin: plugin, type: type }
    end
  end
```

**Review by eviltrout — APPROVED** [review]

**Pull request merged**